### PR TITLE
Feature/api setup

### PR DESCRIPTION
**Pull request by karnthis**
- Adding frames for Ingredient and Shopping (lists)
- Adding TypeORM config
- Adding Swagger

### CODE REVIEW

File: apps/api/src/main.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
   const app = await NestFactory.create(AppModule);
-  await app.listen(process.env.PORT || 3000);
+  const options = new DocumentBuilder()
+    .setTitle('Great Example')
+    .setDescription('The API description')
+    .setVersion('1.0')
+    .build();
+  const document = SwaggerModule.createDocument(app, options);
+  SwaggerModule.setup('swag', app, document);
+  await app.listen(parseInt(process.env.APIPORT, 10));
```
Comment (barbados-clemens) [inline apps/api/src/main.ts]: why port 10? shouldn't it be 80 and/or 443

File: apps/api/src/app.module.ts [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
-  imports: [],
-  controllers: [AppController],
-  providers: [AppService],
+  imports: [IngredientModule, ShoppingModule, FloggerModule,
+    TypeOrmModule.forRoot({
+      type: 'postgres',
+      host: PGHOST,
+      port: parseInt(PGPORT, 10),
+      username: PGUSER,
+      // password: PGPASSWORD,
```
[the remark this answers is not in the document]
Comment (karnthis) [reply apps/api/src/app.module.ts]: Postgres doesn’t require a password on localhost, I left it as a reminder for prod

File: apps/api/src/main.ts (hunk 1 above)
Comment (karnthis) [reply apps/api/src/main.ts]: That isn’t port 10, it is defining that APIPORT is base 10

**Comment by karnthis** [comment]
I have discovered that the dotenv setup in this PR does not work, and requires an additional config I am going to include in the next PR.

File: apps/api/src/main.ts (hunk 1 above)
Comment (spiray) [reply apps/api/src/main.ts]: @karnthis it is generally better practice to use `Number(process.env.APIPORT)` when converting a base 10 number.

Comment (karnthis) [reply apps/api/src/main.ts]: Sure I can change it. I always forget about Number()

**Comment by spiray** [comment]
@karnthis Do you want to include some more in this PR? Like an initial model?

**Comment by karnthis** [comment]
I didn’t want to clutter this too much. I prefer having more, targeted PRs rather than large ones trying to do everything. 

We also need to get a better grasp on what the data model is going to look like. I could see several things in Barb’s mockup front end that I had not planned for.

**Comment by spiray** [comment]
I would also include some of the default values in the .env-sample if the user wont have to change it.

**Comment by karnthis** [comment]
requested changes made for dotenv sample and docker-compose. also fixed missing dotenv config so that will now work post-PR. Thanks for the suggestions @barbados-clemens and @spiray

**Comment by karnthis** [comment]
Added docker scripts (and friends) based on spiray's request

**Comment by barbados-clemens** [comment]
Can we update the README that docker is required for the backend api just as an FYI for anyone that passes by.

**Comment by spiray** [comment]
@barbados-clemens Docker isn't required. I just spun up an ElephantSQL free instance and used those connection values. We should potentially have both methods outlined in the README.

**Pull request merged**